src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -321,6 +333,23 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
                         $params = array_merge($params, $data[$paramName]);
                     }
                 } elseif ($allowed && !$ignored && (isset($data[$key]) || array_key_exists($key, $data))) {
+                    if ($this->propertyInfoExtractor) {
+                        $types = $this->propertyInfoExtractor->getTypes($class, $key);
+
+                        foreach ($types as $type) {
+                            if ($type && $type->getClassName() && (!empty($data[$key]) || !$type->isNullable())) {
+                                if (!$this->serializer instanceof DenormalizerInterface) {
```

Comment:
What benefits would that bring? You still have to verify before each extraction.

```php
$hasExtractor = isset($this->propertyInfoExtractor);

/*...*/

foreach ($properties as $property) {
    if ($hasExtractor) {
    }
}
```

The above is marginally better at the level of one less object-property/hashmap dereference.